Google Cloud Functions documentation (#832)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/google-cloud-functions.mdx

## Getting started

Before configuring Google Cloud Functions as a destination in RudderStack, verify if the source platform is supported by Google Cloud Functions by referring to the table below:

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |  -             | -             | -             |
| **Cloud mode**      |  **Supported** | **Supported** |  **Supported** |

<div class="infoBlock">
To know more about the difference between cloud mode and device mode in RudderStack, refer to the <Link to="/destinations/rudderstack-connection-modes/">RudderStack Connection Modes</Link> guide.
</div>

Once you have confirmed that the source platform supports sending events to Google Cloud Functions, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Google Cloud Functions**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Google Cloud Functions as a destination, you need to configure the following settings:

> <img src="../../assets/event-stream-destinations/gcf-connection-settings.png" alt="Google Cloud Functions connection settings"/>

Author: We will need to update this image to include the newly introduced `maxBatchSize` setting. @RohitKhatta

</div>

- **Gcloud Authorization**: If you have selected **HTTPS** as your trigger type, enter your <a href="https://cloud.google.com/functions/docs/securing/authenticating#authenticating_developer_testing">Google Cloud Authorization Token</a>.
- **URL**: Enter the URL obtained in the **Trigger** field during the cloud function setup.

<div class="infoBlock">
For more information on obtaining the URL, refer to the <Link to="#how-do-i-create-a-google-cloud-function">FAQ</Link> below.
</div>
- **API Key ID**: Enter your API key ID. Refer to the <a href="https://cloud.google.com/docs/authentication/api-keys">Google Cloud documentation</a> for more information on creating and using the API key.

<div class="infoBlock">
The API key ID is a string required by the cloud function to identiy a request. RudderStack injects this key in the <code class="inline-code">ApiKey</code> header in the case of both <strong>HTTP</strong> and <strong>HTTPS</strong> trigger types.
</div>

- **Enable Batch Input**: Enable this setting to pass a batch input (array of events) to your cloud function.

> - **maxBatchSize**: If **Enable Batch Input** setting is enabled, specify the maximum batch size in this field.

Author: Please confirm if `maxBatchSize` is what we are naming the setting. Should it be more user-friendly, for example, `Maximum event batch size` @RohitKhatta cc @ItsSudip

| **Device mode**     |  -             | -             | -             |
| **Cloud mode**      |  **Supported** | **Supported** |  **Supported** |

<div class="infoBlock">
To know more about the difference between cloud mode and device mode in RudderStack, refer to the <Link to="/destinations/rudderstack-connection-modes/">RudderStack Connection Modes</Link> guide.
</div>

Once you have confirmed that the source platform supports sending events to Google Cloud Functions, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Google Cloud Functions**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Google Cloud Functions as a destination, you need to configure the following settings:

<img src="../../assets/event-stream-destinations/gcf-connection-settings.png" alt="Google Cloud Functions connection settings"/>

- **Choose your Trigger Type**: From the dropdown, select **HTTP** or **HTTPS** depending on the trigger type configured for your cloud function.

<div class="infoBlock">

> For more information on setting the trigger type for your cloud function, refer to the <Link to="#faq">FAQ</Link> section below.

Reviewer: not sure if the FAQ provides additional info on what trigger type should be set. It just says choose any based on your use-case. WDYT about removing this infoblock?

Author: the trigger type should be configured while setting up the google cloud function. The info block here acts as a link between this setting and the FAQ

> For more information on setting the trigger type for your cloud function, refer to the <Link to="#faq">FAQ</Link> section below.

Author: ```suggestion
You can set the trigger type while configuring your cloud function. For more information, refer to the <Link to="#faq">FAQ</Link> section below.
```

</div>

- **Gcloud Authorization**: If you have selected **HTTPS** as your trigger type, enter your <a href="https://cloud.google.com/functions/docs/securing/authenticating#authenticating_developer_testing">Google Cloud Authorization Token</a>.
- **URL**: Enter the URL obtained in the **Trigger** field during the cloud function setup.

<div class="infoBlock">
For more information on obtaining the URL, refer to the <Link to="#how-do-i-create-a-google-cloud-function">FAQ</Link> below.
</div>

- **API Key ID**: Enter your API key ID. Refer to the <a href="https://cloud.google.com/docs/authentication/api-keys">Google Cloud documentation</a> for more information on creating and using the API key.

<div class="infoBlock">

> The API key ID is a string required by the cloud function to identiy a request. RudderStack injects this key in the <code class="inline-code">ApiKey</code> header in the case of both <strong>HTTP</strong> and <strong>HTTPS</strong> trigger types.

Reviewer: ```suggestion
The API key ID is a string required by the cloud function to identify a request. RudderStack injects this key in the <code class="inline-code">ApiKey</code> header in case of both the <strong>HTTP</strong> and <strong>HTTPS</strong> trigger types.
```
